test/StreamJsonRpc.Tests/MarshalableProxyTests.cs

```diff
@@ -499,6 +523,40 @@ public async Task DisposeOnDisconnect()
         await disposed.WaitAsync(this.TimeoutToken);
     }
 
+    [Fact]
+    public async Task SubTypeMarshalable_CanCallMethods()
+    {
+        this.server.ReturnedMarshalableWithSubTypes = new MarshalableWithSubTypes();
+        IMarshalableWithSubTypes? proxy = await this.client.GetMarshalableWithSubTypesAsync();
+        Assert.Equal(1, await proxy!.GetAsync(1));
```

Comment:
Asserting that the proxy does _not_ implement certain interfaces here seems prudent.